tools/asmrefine/GraphProof.thy

```diff
@@ -2180,7 +2180,6 @@ lemma stepwise_graph_refine_Basic:
   apply (frule_tac i=i and tr=tr in exec_trace_step_cases)
   apply (frule_tac i=j and tr=tr' in exec_trace_step_cases)
   apply (clarsimp simp: exec_graph_step_def exec_graph_invariant_Cons split: graph_function.split_asm)
-  apply (simp add: K_def)
```

Comment:
does this indicate that the K_def [simp] declaration now leaks everywhere? I saw you put a FIXME to get rid of that declaration... what did you have in mind?

Reply:
It does not fully leak everywhere, yes. It did leak almost everywhere before already, though (it was in most monad definition files, but not all), so I would say this is better than before, because it's at least consistent.

Removing it is a bigger operation and shouldn't be in here. It's not entirely clear yet if it's actually better to remove it or not, but I would hope that it is (depends on how much breaks).